quixstreams/sources/base/manager.py

```diff
@@ -79,6 +111,46 @@ def run(self) -> None:
             "s" if threadcount > 1 else "",
         )
 
+    def _recover_state(self, source: StatefulSource) -> StorePartition:
+        """
+        Recover the state from the changelog topic and return the assigned partition
+
+        For stateful sources only.
+        """
+        recovery_manager = RecoveryManager(
+            consumer=self._consumer,
+            topic_manager=self._topic_manager,
+        )
+
+        state_manager = StateStoreManager(
+            producer=self._producer, recovery_manager=recovery_manager
+        )
+
+        store_name = f"source-{source.store_name}"
```

Comment:
Can this code be moved to the `StatefulSource.store_name` property?
Otherwise the property doesn't really return a store name but only a part of it